### ISSUE TITLE
Inconsistent unmapped but placed query results

### ISSUE THREAD

**Issue by cmnbroad** [broadinstitute/gatk#4901]
When querying bams by interval, htsjdk only returns unmapped, placed reads if the read's start position overlaps the query interval, whereas the read's mapped mate is returned if any part of it overlaps the interval. Unclear if this has an affect on things like finding breakpoint evidence, but it can result in unintentionally splitting up pairs, and can give subtly inconsistent results such as when querying the reads that back a variant, since sometimes a given query can include unmapped reads, sometimes not, and sometimes have a mix.

**Comment by lbergelson** [comment]
This seems really bad.

**Comment by cmnbroad** [comment]
@lbergelson Yeah, it would seem more rational to always keep them together. It would require some lookahead/state to use the same interval for both, since empirically there seems to be no ordering maintained between the reads in a pair.

**Comment by cmnbroad** [comment]
Just noticed that there is an old open ticket for this in htsjdk https://github.com/samtools/htsjdk/issues/278.
